### PR TITLE
Enhance determination of matching interceptor rules in `Predicate` evaluation  

### PR DESCRIPTION
**Pull request by ahennr**
Add if/else branch to evaluate whether interceptor rule is endpoint and service specific.

### CODE REVIEW

**Comment by buehner** [comment]
I tested this successfully with an existing project. But maybe @dnlkoch should have another look at this!?

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java [hunk 1]
```diff
@@ -271,14 +271,26 @@ private InterceptorRule getMostSpecificRule(String requestService,
...
-				// most specific: we have a rule with a matching endPoint
-				//                and operation
+				// most specific: we have a rule with a matching endPoint, service and operation
 				if (Objects.equals(rule.getEndPoint(), endPoint) &&
+						Objects.equals(rule.getOperation(), operation) &&
+						Objects.equals(rule.getService(), service)) {
+					LOG.trace("  * " + rule + " is endPoint, service and operation specific.");
+					return true;
+				// operation specific: if we have a rule with a matching endPoint and service,
+				// but not with a matching operation
```
Comment (marcjansen) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java]: Above comment is misleading, or am I reading the code wrong?

Comment (ahennr) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java]: Alright, fixed in 89d6f6e21

**Comment by buehner** [comment]
ping @dnlkoch

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java [hunk 2]
```diff
@@ -271,14 +271,26 @@ private InterceptorRule getMostSpecificRule(String requestService,
...
+				} else if (Objects.equals(rule.getEndPoint(), endPoint) &&
+						Objects.equals(rule.getService(), service)) {
+					LOG.trace("  * " + rule + " is endPoint and service specific.");
+					return true;
+				// endPoint and operation specific: if we have a rule with a matching endPoint
+				// and operation, but not with a matching service
+				} else if (Objects.equals(rule.getEndPoint(), endPoint) &&
 						Objects.equals(rule.getOperation(), operation)) {
-					LOG.trace("  * " + rule + " is endPoint and operation specific.");
+					LOG.trace("  * " + rule + " is operation and service specific.");
```
Comment (dnlkoch) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java]: …is endPoint and operation specific…

**Comment by dnlkoch** [comment]
LGTM, one minor remark only.

**Comment by ahennr** [comment]
THX for your reviews.

**Pull request merged**